src/helpers/showPopup.ts

```diff
@@ -27,9 +28,10 @@ async function getPosition() {
 export const showPopup = async (chainId: string, event: TransactionEvent) => {
     const { triggerType, requestType, payload } = event;
     const { top, left } = await getPosition();
+    const url = await getActiveTabUrl();
```

Comment:
This make more sense, though it not the most ideal place.

The ideal place to get the URL is from the `content` script which runs in the context of the website itself.
We can have many many content scripts loaded, each one for each page.

The background script is OK since we're assuming user is interacting only with one page at a time
<img width="989" alt="Screenshot 2023-02-28 at 10 14 10" src="https://user-images.githubusercontent.com/120090466/221793221-120aadd4-4b9c-4303-af37-e0a4dc92ccf0.png">

Reply:
Thanks for the diagram :) It didnt work when I tried to query form the attach script, lets land as is for now and we'll try to improve on it in the future